osu.Game/Overlays/Music/Playlist.cs

```diff
@@ -75,10 +71,10 @@ public void Filter(string text)
             {
                 var metadata = playlistItem.BeatmapSetInfo.Metadata;
                 var match = string.IsNullOrEmpty(text)
-                    || (metadata.Artist ?? "").IndexOf(text, StringComparison.InvariantCultureIgnoreCase) != -1
-                    || (metadata.ArtistUnicode ?? "").IndexOf(text, StringComparison.InvariantCultureIgnoreCase) != -1
-                    || (metadata.Title ?? "").IndexOf(text, StringComparison.InvariantCultureIgnoreCase) != -1
-                    || (metadata.TitleUnicode ?? "").IndexOf(text, StringComparison.InvariantCultureIgnoreCase) != -1;
+                            || (metadata.Artist ?? "").IndexOf(text, StringComparison.InvariantCultureIgnoreCase) != -1
+                            || (metadata.ArtistUnicode ?? "").IndexOf(text, StringComparison.InvariantCultureIgnoreCase) != -1
+                            || (metadata.Title ?? "").IndexOf(text, StringComparison.InvariantCultureIgnoreCase) != -1
+                            || (metadata.TitleUnicode ?? "").IndexOf(text, StringComparison.InvariantCultureIgnoreCase) != -1;

```

Comment:
this and a lot of the other logic here is very very very copy-paste. you must always avoid copy pasting out of laziness. if something is used in two places it *needs* to be extracted so it can be reused.